### PR TITLE
Add code coverage badge through coverage.py

### LINKED ISSUE #60: Add test coverage data + badge to project

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR adds a code coverage badge to the readme by running coverage.py through pytests, generating a report, and then generating a badge. Any changes to the coverage should change this file in future PR's.

Additionally:
- I reorganized the logo location to be grouped into a directory where the badge image could be stored under `media`.
- I also took a moment to update pre-commit hooks and related linting in somewhat unrelated files.

Closes #60 

## What kind of change(s) are included?

- [x] Feature (adds or updates new capabilities)
- [ ] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] These changes pass all pre-commit checks.
- [x] I have added comments to my code to help provide understanding
- [ ] I have added a test which covers the code changes found within this PR
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,15 +8,19 @@ repos:
...
       - id: check-added-large-files
       - id: detect-private-key
   - repo: https://github.com/python-poetry/poetry
-    rev: "1.5.1"
+    rev: "1.8.0"
     hooks:
       - id: poetry-check
   - repo: https://github.com/codespell-project/codespell
-    rev: v2.2.6
+    rev: v2.3.0
```
Comment (jenna-tomkinson) [inline .pre-commit-config.yaml:23]: General question, what was your process on deciding to upgrade the versions in this pre-commit file?

File: src/cosmicqc/frame.py [hunk 2]
```diff
@@ -486,7 +486,7 @@ def create_figure_group_html(
         # return the path of the file
         return report_path
 
-    def create_threshold_set_outlier_visualization(  # noqa: PLR0913
+    def create_threshold_set_outlier_visualization(
```
Comment (jenna-tomkinson) [inline src/cosmicqc/frame.py:489]: Why was this removed?

**Review by jenna-tomkinson — APPROVED** [review]
Short and simple PR, nice job! I was wondering why the change from `logo` to `media`, is this a specific standard the packages tend to follow?

I am approving this PR since I don't see any glaring issues, mainly just things I had questions about. Feel free to merge after addressing!

File: src/cosmicqc/frame.py (hunk 2 above)
Comment (d33bs) [reply src/cosmicqc/frame.py:489]: This was an update which seemed to happen as a result of updating the `ruff` pre-commit configuration. It was originally added to avoid a failure with a `ruff` check, and since `ruff` removed it without an error I trust it was the correct update to make.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (d33bs) [reply .pre-commit-config.yaml:23]: Thank you for the question - my process is to consider where I'm making modifications and then try to absorb a bit of additional refactoring with those files / blocks while I'm there to "leave it better than I found it". Some of how I think can be considered ["opportunistic refactoring"](https://martinfowler.com/bliki/OpportunisticRefactoring.html) - we have to be careful about leaving [technical debt](https://en.wikipedia.org/wiki/Technical_debt) alone for too long (we don't know when we'll come back to it!).

**Comment by d33bs** [comment]
Thank you @jenna-tomkinson ! Addressing your question:

> I was wondering why the change from logo to media, is this a specific standard the packages tend to follow?

I think of it like I do modules or classes: when there's one "thing" or "object" it's okay to keep things specific. Once there's more than one "thing" it's useful to create an abstraction to house those "things" in some sort of hierarchy to reduce the amount of "horizontal" spread (really, to reduce the complexity of looking through the repo and its contents). Keep in mind there are contrasting opinions about these things - some prefer nesting with directories and files, others prefer single-directory structures (placing everything into a single directory). I tend to practice nesting, but this doesn't mean it's perfect!

In this case we had one "thing" (the logo code and images). The changes in this PR added another "thing" (the code coverage badge image). As a result I created the abstraction of "media" to help organize these things instead of having two folders for similar but different things. I don't know that there's a standard when it comes to this, I think I picked up on `media` from earlier web framework experiences, but I'm not sure! I have a feeling these may be better suited to go into the docs folder somewhere eventually, but I don't know exactly where yet because we haven't selected a docs framework or strategy (which would inform where it goes).

Thanks again, merging this in!
